app/src/main/java/com/example/concordia_campus_guide/Database/Daos/RoomDao.java

```diff
@@ -28,6 +28,10 @@
     @Query("SELECT * FROM rooms WHERE id=:roomId")
     public RoomModel getRoomById(long roomId);
 
+    @Query("SELECT * FROM rooms WHERE room_code=:roomCode AND floor_code=:floorCode" )
+    public RoomModel getRoomByRoomCodeAndFloorCode(String roomCode, String floorCode);
+
+
```

Comment:
You created a method that already exists. getRoomByIdAndFloorCode takes the same parameters and fetches the same data.

Reply:
oo I didn't noticed that. thank you I will check it out